In [3]:
import pandas as pd
from sklearn.model_selection import train_test_split
from sklearn.ensemble import RandomForestRegressor
from sklearn.metrics import mean_squared_error, r2_score

df = pd.read_csv('cacao_engineered.csv')

#split data 75:25
x = df.loc[:, df.columns != 'Rating']
y = df['Rating']

x_train, x_test, y_train, y_test = train_test_split(x, y, train_size=0.75)

#Random Forest regression with default values
model = RandomForestRegressor()

#fit model
model.fit(x_train, y_train)

#predict on model
y_pred = model.predict(x_test)

#evaluate model
mse = mean_squared_error(y_test, y_pred)
print("Mean Squared Error", mse)

rmse = mse**0.5
print("Root Mean Squared Error", rmse)

r2score = r2_score(y_test, y_pred)
#adjusted rscore is just rscore but the rscore only improves if additional features actually improves the model's performance. 
#Basically it negates the weakness of rscore which is that the score increases as the number of features increase
adjusted_r2score = 1- ((1 - r2score) * ((df.shape[0] - 1) / (df.shape[0] - df.shape[1] - 1)))
print("Adjusted R Squared Score: ", adjusted_r2score)








Mean Squared Error 0.1767164810690423
Root Mean Squared Error 0.42037659434017294
Adjusted R Squared Score:  0.24190951638005953


Above output is from the Random Forest Regression with the default settings. Currently, according to the Adjusted R Squared Score, the model is not performing well since it has a score of 0.24. In other words, the attributes currently do not predict the ratings very well. The Mean Squared Error and Root Mean Squared Error currently don't have much meaning since we don't have other Mean Sqaure Errors and Root Mean Squared Errors to compare them to. So let's try using grid search to see parameters work best for Random Forest Regression.